### ISSUE TITLE
experiment: baseline GRPO run on countdown

### ISSUE THREAD

**Issue by teilomillet** [teilomillet/textpolicy#14]
## Parent Issue
Part of #5 (GRPO optimizations for reasoning emergence)

## Context
See [Aha Moment Research Plan](https://github.com/teilomillet/textpolicy/issues/5) — this is Phase 1.

## Goal

Run vanilla GRPO on countdown task to establish baseline:
1. Does emergence happen at all with current implementation?
2. How many steps until emergence?
3. What's the accuracy curve?
4. What does the model output look like over time?

### Model
- **Primary:** Qwen3-1.5B-Instruct (minimum viable for emergence)
- **Fallback:** Qwen3-0.6B (if memory constrained)

### Task
- Countdown game (#12)
- Dataset: Jiayi-Pan/Countdown-Tasks-3to4

### Training Config
```yaml
model: Qwen/Qwen3-1.5B-Instruct
algorithm: grpo
learning_rate: 5e-6
batch_size: 8
num_generations_per_prompt: 4
max_steps: 500
max_completion_length: 512
clip_ratio: 0.2  # Symmetric (baseline)
```

### Logging
- Use EmergenceLogger (#13)
- Log all generations
- Track planning phrase ratio per step

### Positive Signals (Emergence Happening)
- [ ] Planning phrases appear that weren't in prompt
- [ ] Accuracy improves over training
- [ ] Completion length increases (more reasoning)
- [ ] "Wait", "let me try", "actually" start appearing

### Negative Signals (Problems)
- [ ] Entropy collapses (need decoupled clipping)
- [ ] Accuracy doesn't improve (model too small?)
- [ ] Generations become repetitive (need more exploration)
- [ ] Training unstable (need stability fixes)

## Litmus Test

| Metric | Baseline Threshold | Notes |
|--------|-------------------|-------|
| **Any emergence** | Planning phrases appear by step 500 | Primary success criterion |
| **Accuracy** | > 10% by step 500 | Some learning happened |
| **Entropy** | > 0.5 at step 500 | Not collapsed |

### Deliverables
1. Training logs (`generations.jsonl`, `steps.jsonl`)
2. Accuracy curve plot
3. First emergence step (if any)
4. Example generations at step 0, 100, 300, 500
5. Assessment: proceed with optimizations or investigate issues

### Decision Tree

```
IF emergence happens:
  → Document timing, move to Phase 2 (stability) to improve
  
IF no emergence but training stable:
  → Add GTPO (#10), try again
  
IF training unstable (entropy collapse):
  → Add decoupled clipping (#6), try again
  
IF model too small:
  → Try larger model or different task
```

## Dependencies

- #12 Countdown task implementation
- #13 Generation logging system

## Run Command

```bash
# Once implemented
textpolicy train \
  --config experiments/countdown_baseline.yaml \
  --output results/baseline_001
```
